File: SonarQube.CSharp/SonarQube.Rules.Test/TestCases/ConditionalStructureSameImplementation_If.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Tests.TestCases
{
    class ConditionalStructureSameCondition_If
    {
        public bool someCondition1 { get; set; }
        public bool someCondition2 { get; set; }
        public bool someCondition3 { get; set; }

        public void DoSomething1() { }
        public void DoSomething2() { }

        public void Test()
        {
            if (someCondition1)
            {
                DoSomething1();
            }
            else
            { // Noncompliant
                DoSomething1();
            }

            if (someCondition1)
            {
                DoSomething1();
            }
            else if (someCondition2)
            { // Noncompliant     
                DoSomething1();
            }
            else if (someCondition3)
            {
                DoSomething2();
            }
            else
            { // Noncompliant
                DoSomething1();
            }

            if (someCondition1)
            {
                DoSomething1();
            }
            else if (someCondition2)
            { // Noncompliant     
                DoSomething1();
            }
            else
            {// Noncompliant 
                DoSomething1();
            }
        }
    }
}

File: sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/ConditionalStructureSameImplementation_If.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace Tests.TestCases
    Class ConditionalStructureSameCondition_If
        Sub Test()
            If (someCondition1) Then
                DoSomething1()
            Else
                DoSomething1() ' Noncompliant
            End If

            If someCondition Then DoSomething1() Else DoSomething1() End ' Noncompliant
'                                                     ^^^^^^^^^^^^^^
            If someCondition Then DoSomething1() Else DoSomething2() : DoSomething1() End

            If (someCondition1) Then
                DoSomething1()
                DoSomething1()
            ElseIf (someCondition2) Then

                DoSomething1()
                DoSomething2()
            ElseIf (someCondition2) Then

                DoSomething1() ' Noncompliant {{Either merge this branch with the identical one on line 25 or change one of the implementations.}}
                DoSomething2()
            ElseIf (someCondition2) Then

                DoSomething1() ' Noncompliant {{Either merge this branch with the identical one on line 25 or change one of the implementations.}}
                DoSomething2()
            Else
                DoSomething1() ' Noncompliant {{Either merge this branch with the identical one on line 21 or change one of the implementations.}}
                DoSomething1()
            End If

            If (someCondition1) Then
                DoSomething1()
            ElseIf (someCondition2) Then
                DoSomething1() ' Noncompliant
            Else
                DoSomething1() ' Noncompliant
            End If
        End Sub
    End Class
End Namespace
